src/Symfony/Component/Form/FormFactory.php

```diff
@@ -73,6 +73,7 @@ public function createBuilder($type = 'form', $data = null, array $options = arr
     public function createNamedBuilder($name, $type = 'form', $data = null, array $options = array())
```

Comment:
To make the feature forward compatible, you should allow one to pass $options as second argument.
Something like:
```php
if (is_array($data) && 2 === func_num_args()) {
    $options = $data;
    $data = null;
}
```